src/containers/pages/OrganizationViews/AssignPractitioners/index.tsx

```diff
@@ -0,0 +1,279 @@
+  Organization,
+  reducerName as organizationReducerName,
+} from '../../../../store/ducks/opensrp/organizations';
+import practitionersReducer, {
+  fetchPractitionerRoles,
+  getPractitionersByOrgId,
+  Practitioner,
+  reducerName as practitionerReducerName,
+} from '../../../../store/ducks/opensrp/practitioners';
+import { loadOrganization, loadOrgPractitioners } from '../serviceHooks';
+import { filterOptions, formatOptions, SelectOption, styles } from './utils';
+
+reducerRegistry.register(organizationReducerName, organizationsReducer);
+reducerRegistry.register(practitionerReducerName, practitionersReducer);
+
+/** Props for AssignPractitioner component */
+interface AssignPractitionerProps {
+  fetchOrganizationsCreator: typeof fetchOrganizations;
+  fetchPractitionerRolesCreator: typeof fetchPractitionerRoles;
+  organization: Organization | null;
+  serviceClass: typeof OpenSRPService;
+  assignedPractitioners: Practitioner[];
+}
+
+/** default props for AssignPractitioner component */
+const defaultAssignPractitionerProps: AssignPractitionerProps = {
+  assignedPractitioners: [],
+  fetchOrganizationsCreator: fetchOrganizations,
+  fetchPractitionerRolesCreator: fetchPractitionerRoles,
+  organization: null,
+  serviceClass: OpenSRPService,
+};
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = AssignPractitionerProps & RouteComponentProps<RouteParams>;
+
+/** AssignPractitioner component */
+const AssignPractitioner: React.FC<PropsTypes> = props => {
+  const {
+    serviceClass,
+    fetchOrganizationsCreator,
+    organization,
+    fetchPractitionerRolesCreator,
+    assignedPractitioners,
+  } = props;
+  const [selectedOptions, setSelectedOptions] = useState<OptionsType<SelectOption>>([]);
+
+  useEffect(() => {
+    const organizationId = props.match.params.id;
+    loadOrganization(organizationId, serviceClass, fetchOrganizationsCreator);
+    loadOrgPractitioners(organizationId, serviceClass, fetchPractitionerRolesCreator);
+  }, []);
+
+  if (!organization) {
+    return <Loading />;
+  }
+
+  // TODO - scenario where this request doesn't return all practitioners in a single query
+  /** load all practitioners at least all of those returned
+   * in a single call to Practitioners endpoint
+   * @param {typeof OpenSRPService} service - the OpenSRPService
+   */
+  const loadAllPractitioners = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const serve = new service(OPENSRP_PRACTITIONER_ENDPOINT);
+    const allPractitioners = await serve.list();
+    return formatOptions(allPractitioners);
+  };
+
+  // TODO - Hack in this: typings for the changeHandler function.
+  /** This sets the state selectedOptions
+   * @param {ValueType<SelectOption>} -  the so far selected options
+   * @param {ActionMeta} - information on the change event; custom react-select event
+   */
+  const changeHandler = (chosenOptions: any, { action, removedValue, option }: any) => {
```

Comment:
Why the `any` hack and not the imported Typings?

Reply:
I believe the typings for this changeHandler are not in sync with the library code
the allowed type i.e [ActionMeta](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/643edf5b1c8a4e32d2f5a1e61cfa64dac6ba098d/types/react-select/src/types.d.ts#L69) for the second argument  to the changeHandler only accounts for the `action`, but doesn't for the removedValue, or option

Reply:
What do you think needs to be done to resolve the "hack" - should we create an issue with those details?